### ISSUE TITLE
Config file / folder structure bug

### ISSUE THREAD

**Issue by Thyriel** [RoMiMoCo/Ores#17]
There seems to be a minor bug with where config is stored. After first start it created the "ores.cfg" but no "ores" folder. I created that folder then and placed "oreDefinitions.json" inside that. After next restart all Files where moved to "Romimoco/ores/ores" and a new config file created (so there's 2 now but only the ores.cfg inside ores/ores seems valid)

Also the language file is generated, but does not work. (Still having no names ingame)

Imho as you're only using 1 OreDefinitions.json and not reading one file per ore it would be better just placing all files in the root of config/Romimoco instead using subfolders inside that.

**Comment by Thyriel** [comment]
I know i'm getting nasty, but could you maybe add a config option too to turn off the ore item ? Would be nice to add ingots / blocks for alloys without any ore being added

**Comment by RoMiMoCo** [comment]
The config directory structure should be config/Romimoco/ores.  This is because I will eventually be adding more mods to the collection and want to be keep all of the config files together.  I'll look into why the subdirectory isn't being created, as it should be.

The language file needs to be put into a resource pack in order for it to work, so in the resourcepacks folder there needs to be a resourcepackname/assets/minecraft/lang directory where that file goes.  And in the top level of that resourcepack there needs to be a pack.mcmeta that contains at a minimum {"pack":{"pack_format":3,"description":"Fixing names for romimoco-ores"}}.  This is still a manual process for now.

The ore item needs to stay (although.... I could not just register it.... thinking), as all of the other items are generated based upon that ore.  However, if there is no "Generation" section in OreDefinitions it will not be added to world generation.  So the ore item would exist in creative mode and NEI, but there would be no way to obtain it.

**Comment by RoMiMoCo** [comment]
Oh, and thanks for the feedback.  Having more eyes on possible bugs makes it easier to track them down.

**Comment by Thyriel** [comment]
Should the ores.cfg be in the root of the config or inside the "ores" directory ? Cause currently it creates both and seems to read only the one inside "ores".

Language file is working this way, although it's not a neat solution, especially when giving the modpack to someone they always need to manually activate the resourcepack. (Probably look at Infini-Tic which is opensource how they did it.)

Removing oregen section worked. (Although i would still prefer to not have the ore item as it may confuse players to look for the ore when they want to make that alloy but i'll just name the ore something like dummy or so then)

**Comment by RoMiMoCo** [comment]
ores.cfg should be in the ores subdirectory.  I'm not sure why it creates the other.

**Comment by Thyriel** [comment]
> if there is no "Generation" section in OreDefinitions it will not be added to world generation

Just seen in my world, if the "Generation" section is completely removed the ore still spawns above bedrock height in huge veins.
Setting SpawnChance to 0 and VeinSize to 0 worked.

**Comment by RoMiMoCo** [comment]
Yep, was able to replicate that.  Was missing a null check, so it was falling through to the defaults I had set up for if it couldn't parse the json object.  Fixed in 0.2.1

I cannot seem to replicate the issues you had with the .cfg file.  On a clean install of forge with only ores installed, it creates the Romimoco/ores directory structure correctly.

I'm going to break out the language file and the worldgen issues into seperate enhancement requests to better track them.
